### PR TITLE
Mgmt: Fix ProviderRegistrationPolicy using a existing Providers endpoint

### PR DESCRIPTION
**Pull request by ChenTanyi**
https://github.com/Azure/azure-sdk-for-java/issues/16805

First add a minor skip. Final fix would be regenerate pipeline for `withSubscription` and `withDefaultSubscription`

### CODE REVIEW

**Review by weidongxu-microsoft — APPROVED** [review]

**Review by xccc-msft — APPROVED** [review]

**Review by andrewchoi5 — APPROVED** [review]

**Comment by ChenTanyi** [comment]
After change the API, it may lead to breaking change, which cannot pass the CI for PR and Release.

File: sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/HttpPipelineProvider.java [hunk 1]
```diff
@@ -73,7 +72,6 @@ public static HttpPipeline buildHttpPipeline(
         if (credential != null) {
             policies.add(new AuthenticationPolicy(credential, profile.getEnvironment(), scopes));
         }
-        policies.add(new ProviderRegistrationPolicy(credential, profile));
```
Comment (weidongxu-microsoft) [inline sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/HttpPipelineProvider.java:76]: Is it possible to still keep it (automatically set the provider, after auth done)?

It still provide a function that user sometimes needs (hence we got report on the bug).

Comment (ChenTanyi) [reply sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/HttpPipelineProvider.java:76]: If we want, we could set it inside `ResourceManager` constructor. If so, the `AuthorizationManager` will not has such function by default.

Comment (ChenTanyi) [reply sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/HttpPipelineProvider.java:76]: Done, without `AuthorizationManager`

**Review by weidongxu-microsoft — APPROVED** [review]

File: sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/HttpPipelineProvider.java (hunk 1 above)
Comment (weidongxu-microsoft) [reply sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/HttpPipelineProvider.java:76]: Great. Remember to test it a bit (with some uncommon provider).

Comment (ChenTanyi) [reply sdk/resourcemanager/azure-resourcemanager-resources/src/main/java/com/azure/resourcemanager/resources/fluentcore/utils/HttpPipelineProvider.java:76]: Also added a check for null providers.

**Comment by ChenTanyi** [comment]
My concern is I don't want to set `providers` to a policy that user has already set it, so check the `providers` first.

**Pull request merged**